atd-vze/src/views/Locations/LocationEditMap.js

```diff
@@ -0,0 +1,224 @@
+import React, { Component } from "react";
+import MapGL, { NavigationControl, FullscreenControl } from "react-map-gl";
+import { Editor, EditorModes } from "react-map-gl-draw";
+import { withApollo } from "react-apollo";
+import styled from "styled-components";
+
+import Toolbar from "./toolbar";
+import { Button, ButtonGroup } from "reactstrap";
+import { UPDATE_LOCATION } from "../../queries/Locations";
+
+const TOKEN = process.env.REACT_APP_MAPBOX_TOKEN;
+
+const MapStyleSelector = styled.div`
+  top: 24px;
+  right: 24px;
+  position: absolute;
+`;
+
+const fullscreenControlStyle = {
+  position: "absolute",
+  top: 0,
+  left: 0,
+  padding: "10px",
+};
+
+const navStyle = {
+  position: "absolute",
+  top: 36,
+  left: 0,
+  padding: "10px",
+};
+
+class LocationEditMap extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      // map
+      viewport: {
+        width: 800,
+        height: 600,
+        longitude: this.props.data.atd_txdot_locations[0].longitude,
```

Comment:
potential code cleanliness suggestion. Go ahead and make `this.props.data.atd_txdot_locations[0]` a variable since you're using it a bunch.

```js
const location = this.props.data.atd_txdot_locations[0]
```

Reply:
Agreed, I made this change and also added a isSubmitting boolean to change the text in the Submit button as a first step to a better loading indicator.